src/main/java/in/projecteka/gateway/registry/RegistryService.java

```diff
@@ -31,11 +32,34 @@
     private final AdminServiceClient adminServiceClient;
 
     public Mono<ClientResponse> populateCMEntry(CMServiceRequest request) {
-        return registryRepository.getCMEntryIfActive(request.getSuffix())
-                .flatMap(cmEntry -> cmEntry.isExists()
-                        ? updateCMEntry(cmEntry, request)
-                        : createCMEntry(request)
-                );
+        return Mono.just(request)
+                .filterWhen(this::validateRequest)
+                .flatMap(req -> updateCMRequest(request))
+                .flatMap(updatedRequest -> registryRepository.getCMEntryIfActive(updatedRequest.getSuffix())
+                        .flatMap(cmEntry -> cmEntry.isExists()
+                                ? updateCMEntry(cmEntry, updatedRequest)
+                                : createCMEntry(updatedRequest)
+                        ));
+    }
+
+    private Mono<CMServiceRequest> updateCMRequest(CMServiceRequest request) {
+        var updateCMServiceRequest = request.toBuilder();
+
+        if (request.getIsActive() == null)
+            updateCMServiceRequest.isActive(true);
+        if (request.getIsBlocklisted() == null)
+            updateCMServiceRequest.isBlocklisted(false);
+
+        return Mono.just(updateCMServiceRequest.build());
+    }
+
+    private Mono<Boolean> validateRequest(CMServiceRequest request) {
+        if (request != null
+                && ((request.getSuffix() == null || request.getSuffix().isBlank())
+                || (request.getUrl() == null || request.getUrl().isBlank())) )
+            return Mono.error(invalidCMRegistryRequest());
+
+        return Mono.just(true);
```

Comment:
you can validate this in request body itself by using @valid & @notEmpty/@notNull annotations

Reply:
True. But we want to validate if the URL and suffix with empty string/white spaces.

Reply:
yeah.. to do that, we have @NotEmpty & @NotBlank annotations

Reply:
Yeah. But we want to throw Error as a bad request with a valid message. Instead it throws internal server error if the the validation fails. Can we also handle that as well??